File: src/jenkem/shared/ConversionMethod.java
package jenkem.shared;

/**
 * Enum for different ASCII-conversion methods.
 */
public enum ConversionMethod {
    SuperHybrid("Super-Hybrid"),
    FullHd("Full HD"),
    Pwntari("Pwntari"),
    Hybrid("Hybrid"),
    Plain("Plain");

    private String name;

    private ConversionMethod(final String name) {
        this.name = name;
    }

    public static ConversionMethod getValueByName(final String name) {
        for (final ConversionMethod method : ConversionMethod.values()) {
            if (method.getName().equalsIgnoreCase(name)) {
                return method;
            }
        }
        return SuperHybrid;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: src/main/scala/jenkem/engine/ConversionMethod.scala
package jenkem.engine

object ConversionMethod {
  sealed class Value(val name: String)
  case object Vortacular extends Value("Vortacular")
  case object Plain extends Value("Plain")
  case object Stencil extends Value("Stencil")
  val values = List(Vortacular, Plain, Stencil)

  def valueOf(name: String): Option[Value] = values.find(_.name.equalsIgnoreCase(name))
}
